onboarding-enabler-java/src/main/java/org/zowe/apiml/eurekaservice/client/util/EurekaInstanceConfigCreator.java

```diff
@@ -27,21 +27,14 @@
 
 public class EurekaInstanceConfigCreator {
 
-    public EurekaInstanceConfig createEurekaInstanceConfig(ApiMediationServiceConfig config) throws ServiceDefinitionException {
+    public EurekaInstanceConfig createEurekaInstanceConfig(ApiMediationServiceConfig config) throws ServiceDefinitionException, MalformedURLException {
+        EurekaInstanceConfigValidator eurekaInstanceConfigValidator = new EurekaInstanceConfigValidator();
+        eurekaInstanceConfigValidator.validate(config);
         ApimlEurekaInstanceConfig result = new ApimlEurekaInstanceConfig();
 
-        String hostname;
-        int port;
-        URL baseUrl;
-
-        try {
-            baseUrl = new URL(config.getBaseUrl());
-            hostname = baseUrl.getHost();
-            port = baseUrl.getPort();
-        } catch (MalformedURLException e) {
-            String message = String.format("baseUrl: [%s] is not valid URL", config.getBaseUrl());
-            throw new ServiceDefinitionException(message, e);
-        }
+        URL baseUrl = new URL(config.getBaseUrl());
```

Comment:
I would catch the Exception here and then throw a RuntimeException as the failure should never happen here